conn_go19.go

```diff
@@ -0,0 +1,28 @@
+// +build go1.9
+
+package pq
+
+import (
+	"database/sql/driver"
+	"reflect"
+)
+
+func (c *conn) CheckNamedValue(nv *driver.NamedValue) (err error) {
+	if _, ok := nv.Value.(driver.Valuer); ok {
+		// Ignore Valuer, for backward compatiblity with pq.Array()
+		return driver.ErrSkip
+	}
+
+	v := reflect.ValueOf(nv.Value)
+	switch v.Kind() {
+	case reflect.Array, reflect.Slice:
+		// Ignoring []byte / []uint8
+		if v.Type().Elem().Kind() == reflect.Uint8 {
+			return driver.ErrSkip
+		}
```

Comment:
Any reason not to do
```go
if _, ok := nv.Value.([]unit8); ok {
    return driver.ErrSkip
}
```
above? That would spare an indentation level, simplify the condition, and would probably perform better (although it isn't important here). Besides, then you could replace `switch` with a simple
```go
if k := v.Kind(); k == reflect.Array || k == reflect.Slice {
    nv.Value, err = Array(nv.Value).Value()
    return err
}
```